pkgs/ffigen/test/native_objc_test/arc_test.dart

```diff
@@ -97,6 +92,10 @@ void main() {
       expect(objectRetainCount(obj2raw), 3);
       expect(objectRetainCount(obj3raw), 2);
 
+      expect(obj1, isNotNull); // Force obj1 to stay in scope.
```

Comment:
This would not be needed if the objects are `Finalizable`, but objects being `Finalizable` make them non-sharable. 
So maybe we need the "`StayAlive`" marker interface @dcharkes!

In JNIgen I don't have a direct `.pointer` getter, instead users have to write `.reference.pointer` maybe that fixes the issue? I wonder if that works as `.reference` is used in the scope and is `Finalizable` but `Pointer` is not `Finalizable`.

Reply:
I'm not worried about it. This is only an issue in these tests, where I'm explicitly testing the GC behavior. I don't think this will matter to users.

Reply:
I am worried about it lol! We had this GC problem in `package:jni` that happened in user code randomly.

Reply:
Interesting. How did it manifest for users? I haven't been able to think of a situation where it would matter, other than tests like this.